client/boot/app.js

```diff
@@ -0,0 +1,59 @@
+// Initialize localStorage polyfill before any dependencies are loaded
+require( 'lib/local-storage' )();
+if ( process.env.NODE_ENV === 'development' ) {
+	require( 'lib/wrap-es6-functions' )();
+}
+
+/**
+ * External dependencies
+ */
+import * as common from './common';
+import debugFactory from 'debug';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import createReduxStoreFromPersistedInitialState from 'state/initial-state';
+import detectHistoryNavigation from 'lib/detect-history-navigation';
+import userFactory from 'lib/user';
+
+const debug = debugFactory( 'calypso' );
+
+const withProject = () => {
+	// TODO: make project name dynamic based on config
+	const project = require( './project/wordpress-com' );
+
+	return ( funcName, ...params ) => {
+		common[ funcName ]( ...params );
```

Comment:
I like the following idea:
```js
const invoker = ( ctx ) => ( ...args ) => _.invoke( ctx, ...args );
const invoke2 = _.over( [ common, project ].map( invoker ) );
invoke2( 'locales', 'hellooo' );
```
I'm only worried that it would make it more difficult to discover that we try to call the same method from `common` and from `project` files. At the moment it's easy to see what we are doing:
```js
locales( currentUser );
invoke( project, 'locales', currentUser );
```